sonaranalyzer-dotnet/src/SonarAnalyzer.Common/Helpers/SymbolHelper.cs

```diff
@@ -262,5 +262,17 @@ public static bool IsDestructor(this IMethodSymbol method)
         {
             return method.MethodKind == MethodKind.Destructor;
         }
+
+        internal static bool IsKnownType(this SyntaxNode syntaxNode, KnownType knownType, SemanticModel semanticModel)
+        {
+            var symbolType = semanticModel.GetSymbolInfo(syntaxNode).Symbol.GetSymbolType();
+            return symbolType.Is(knownType);
+        }
+
+        internal static bool IsDeclarationKnownType(this SyntaxNode syntaxNode, KnownType knownType, SemanticModel semanticModel)
+        {
+            var symbolType = semanticModel.GetDeclaredSymbol(syntaxNode)?.GetSymbolType();

```

Comment:
perhaps the `?` is not needed here - `GetSymbolType` will handle the null value?